Make changing the testing framework after setting options an error (#22177)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

Related build errors can be avoided by making sure that language version is set on all toolchain specifications.
See <<toolchains#sec:configuring_toolchain_specifications,user manual>> for more information.

==== Changing test framework after setting test framework options is now an error

> When configuring the built-in test task for Java/Groovy/Scala projects, Gradle no longer allows you to

Reviewer: ```suggestion
When configuring the built-in test task for Java, Groovy, and Scala projects, Gradle no longer allows you to
```
nitpick: slashes don't feel like natural parts of a sentence, and tend to disrupt reading. Consider switching to commas.

> change the test framework used by the `Test` task after configuring options. This was deprecated and silently
> discarded configuration in some cases.

Reviewer: ```suggestion
change the test framework used by the `Test` task after configuring options. This was deprecated since it silently
discarded configuration in some cases.
```
Suggestion: This doesn't seem like a natural "and" usage to me. I *think* "silently discarded configuration" is actually the reason this feature was deprecated... so we should make that obvious!

> This now produces an error:

Reviewer: ```suggestion
The following code example now produces an error:
```
Suggestion: avoid "this" pronoun in favor of a more explicit reference to what "this" is.

```
test {
   options {
   }

   useJUnitPlatform()
}
```

> You should either set the test framework before configuring options or look into migrating to the <<jvm_test_suite_plugin#jvm_test_suite_plugin,JVM Test Suite Plugin>>.

Reviewer: ```suggestion
Instead, you can:

- set the test framework before configuring options
- migrate to the <<jvm_test_suite_plugin#jvm_test_suite_plugin,JVM Test Suite Plugin>>
```

Suggestion: convert this list of alternatives to a bulleted list for visibility and scannability. Introduce that list with "instead" to make it more obvious that this is an alternate example to the previous counterexample.
